perun/postprocess/moving_average/methods.py

```diff
@@ -0,0 +1,221 @@
+"""
+Module with computational method of moving average and auxiliary methods at executing of this method.
+"""
+import click
+import numpy as np
+import pandas as pd
+import sklearn.metrics
+
+import perun.postprocess.regression_analysis.tools as tools
+
+# required precision for moving average models when the window width was not entered
+_MIN_R_SQUARE = .88
+# increase of the window width value in each loop, if it has not been reached the required precision
+_WINDOW_WIDTH_INCREASE = .15
+# starting window width as the part of the length from the whole current interval
+_INTERVAL_LENGTH = .05
+
+
+def get_supported_decay_params():
+    """Provides all names of currently supported parameters to specify
+    the relationship to determine the smoothing parameter at Exponential
+    Moving Average.
+
+    :returns list of str: the names of all supported parameters
+    """
+    return list(_DECAY_PARAMS_INFO.keys())
+
+
+def window_width_change(window_width, r_square):
+    """
+    Computation of window width change based on the difference of required and current
+    coefficient of determination (R^2) and possible change of the window width.
+
+    The minimal value of the resulting change cannot be less than 1.
+    The maximal value of the resulting change cannot be greater than 90% of current window_width.
+    The new value of the window with cannot be less than 1.
+    The formula to determine the change of the window width depends on the three values:
+        - difference between coefficient of determinations: required_R^2 - current_R^2
+        - possible change of the window width (minimal width is equal to 1): window_width - 1
+        - value of the increase constant of the window width: WINDOW_WIDTH_INCREASE
+    The resulting change of the window width is the product of multiplication of these values.
+
+    :param int window_width: current window width from the last run of computation
+    :param float r_square: coefficient of determination from the current moving average model
+    :return int: new value of window width for next run of iterative computation
+    """
+    return max(1, window_width - max(1, int(
+        min(.9 * window_width - 1, _WINDOW_WIDTH_INCREASE * (window_width - 1) * (_MIN_R_SQUARE - r_square)))))
+
+
+def compute_moving_average(data_gen, configuration):
+    """
+    The moving average wrapper to execute the analysis on the individual chunks of resources.
+
+    :param iter data_gen: the generator object with collected data (data provider generators)
+    :param dict configuration: the perun and option context
+    :return: list of dict: the computation results
+    """
+    # checking the presence of specific keys in individual methods
+    tools.validate_dictionary_keys(configuration, _METHOD_REQUIRED_KEYS[configuration['moving_method']], [])
+
+    # list of result of the analysis
+    analysis = []
```

Comment:
Suggestion: I would consider renaming the "analysis" variable to "models" or something more appropriate.